services/digest_utils.py

```
@@ -0,0 +1,251 @@
+                    date_part = parts[2]  # YYYY-MM-DD
+                    filename = parts[3]
+                    
+                    # Parse date and convert to YYYY/MM/DD format
+                    date_obj = datetime.strptime(date_part, "%Y-%m-%d")
+                    year = str(date_obj.year)
+                    month = f"{date_obj.month:02d}"
+                    day = f"{date_obj.day:02d}"
+                    
+                    # Construct intro PNG filename
+                    intro_png = f"{filename}_01_intro.png"
+                    return f"https://{self.worker_domain}/assets/stories/{year}/{month}/{day}/{intro_png}"
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)